src/types.rs

```diff
@@ -0,0 +1,139 @@
+pub const EKM_LABEL: &[u8; 21] = b"EXPORTER-pluto-notary";
+
+/// Context used for exporting key material in the TEE attestation process.
+pub const EKM_CONTEXT: &[u8; 3] = b"tee";
+
+/// A struct representing a dummy token used for testing purposes.
+///
+/// The `DummyToken` struct is used to generate TEE attestation tokens by providing a predefined token value.
+/// It contains a single field, `body`, which holds the token value as a string.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DummyToken {
+    /// The token value that the generator will use to create the TEE attestation token.
+    pub body: String,
+}
+
+/// A struct representing a JSON Web Token (JWT) used for TEE attestation.
+///
+/// The `JwtToken` struct contains various fields that represent the claims and attributes
+/// of a JWT used in the TEE attestation process.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JwtToken {
+    /// The audience claim identifies the recipients that the JWT is intended for.
+    pub aud: String,
+    /// The expiration time claim identifies the expiration time on or after which the JWT must not be accepted for processing.
+    pub exp: u64,
+    /// The issued at claim identifies the time at which the JWT was issued.
+    pub iat: u64,
+    /// The issuer claim identifies the principal that issued the JWT.
+    pub iss: String,
+    /// The not before claim identifies the time before which the JWT must not be accepted for processing.
+    pub nbf: u64,
+    /// The subject claim identifies the principal that is the subject of the JWT.
+    pub sub: String,
+    /// The nonce used in the Entity Attestation Token (EAT) for preventing replay attacks.
+    pub eat_nonce: EatNonce,
+    /// The profile used in the EAT.
+    pub eat_profile: String,
+    /// A boolean indicating whether secure boot is enabled.
+    pub secboot: bool,
+    /// The OEM ID of the device.
+    pub oemid: u32,
+    /// The hardware model of the device.
+    pub hwmodel: String,
+    /// The name of the software running on the device.
+    pub swname: String,
+    /// The version of the software running on the device.
+    pub swversion: Vec<String>,
+    /// The attester's Trusted Computing Base (TCB) information.
+    pub attester_tcb: Vec<String>,
+    /// The debug status of the device.
+    pub dbgstat: String,
+    /// The submodules included in the JWT.
+    pub submods: SubModules,
+    /// The Google service accounts associated with the JWT.
+    pub google_service_accounts: Vec<String>,
+}
+
+/// An enum representing the nonce used in the Entity Attestation Token (EAT).
+///
+/// The `EatNonce` enum can either be a single string value or multiple string values.
+/// This is used to prevent replay attacks in the TEE attestation process.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum EatNonce {
+    /// A single nonce value.
+    Single(String),
+    /// Multiple nonce values.
+    Multiple(Vec<String>),
+}
```

Comment:
Why does having this enum help? Couldn't you just have a vec of length 1 satisfy the `EatNonce::Single(_)` variant?

Reply:
Google returns different JSON blobs, either String or Vec<String>.